lib/RGBot.js

```diff
@@ -30,18 +34,105 @@ const { Vec3 } = require('vec3');
  *  <b><u>Item</u></b><br>
  *    An Item represents any Entity that can be collected in the player's inventory or hands.
  *    These can be things like weapons and armor that the player equips, crafting materials, or items that can be placed to create a Block.
- *    This last example brings up an important distinction to keep in mind while creating your Bot: an object is an Item when in the Bot's inventory or hand, or when it has been tossed on the ground, but it is a Block once it is placed in the world.
+ *    This last example brings up an important distinction to keep in mind while creating your Bot: an object is an Item when in the bot inventory or hand, or when it has been tossed on the ground, but it is a Block once it is placed in the world.
  *
  *  <b><u>Name versus Display Name</u></b><br>
  *    An Entity's name is a unique identifier, and its display name is typically the same or similar identifier but in a human-readable format.
  *    As an example, the Ender Dragon is the readable name, or display name, of the Entity named ender_dragon. Likewise, Grass Block is the display name of the block named grass_block.
- *    This library attempts to accept both the name and display name interchangeably wherever possible, so the identifier you use is up to your own personal tastes.
+ *    This library provides functions to accept the name but not the display name for conciseness and efficiency
  */
 const RGBot = class {
 
+    /**
+     * This value is read by the handlePath function to know if the bot is busy using a container while evaluating if it is stuck or not.
+     *
+     * If you open a container type that isn't supported yet and use something other than the provided bot.openContainer(chestBlock) and  bot.closeContainer(containerWindow) methods, you should follow the example.
+     *
+     * @example
+     * let openChest = await bot.openContainer(chestBlock) // this sets bot.isLookingInContainer = true
+     * if (openChest) {
+     *     try {
+     *         //  take action on the container
+     *     } finally {
+     *         try {
+     *             await openChest.close()
+     *         } finally {
+     *             bot.isLookingInContainer = false;
+ *             }
+     *     }
+     * }
+     *
+     * @type {boolean}
+     */
+    isLookingInContainer = false;
```

Comment:
- I wanted these to be visible to players if they need to choose actions based on them
- We don't currently support anywhere close to all the different interaction types we need to.  Thus, this is exposed / documented so that devs can know how not to get spammed with 'your bot is stuck'.